---
title: "Results"
author: "Robert Schwarz"
date: "2023-07-11"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.


### CORRELATION MATRIX EXPRESSION X ACHILLES

The correlation matrix between prism.exp and prism.achilles was done on the common columns of the two, as we are only interested on the genes that both data frames cover and so provide information for
```{r}
#We cannot calculate a correlation because the dimensions of the matrices do not match!
#We have to perform the correlation only on the common genes
common_genes_exp_ach = intersect(colnames(new_prism.exp), colnames(new_prism.achilles))

# We will only include the common genes in our analysis
cor_of_prism.exp = new_prism.exp[, common_genes_exp_ach]
cor_of_prism.achilles = new_prism.achilles[, common_genes_exp_ach]
```

Now this has to be run in a parallel panel, as it takes very long to finish (~4h !)
```{r}
# Because this chunk takes very long, i want to track the progress of it
# Determine the total number of iterations
total_number_iterations = length(common_genes_exp_ach)

# Now we will compute the correlation matrix for the common genes from the data frames achilles and expression
#This correlation creates the link between gene expression and gene relevance! If the anti correlation is strong, it means that the genes taht are relevant for cell proliferation are also highly expressed and the genes less relevant also have a lower expression!
#Let's first fill thsi matrix with NAs just tog et the dimensions right. Then we will fill it up with actual values hehehe
correlation_matrix_exp_ach = matrix(NA, nrow = total_number_iterations, ncol = total_number_iterations,
                                     dimnames = list(common_genes_exp_ach, common_genes_exp_ach))

# Let's fill it up with correlation values! This is where the fun begins...
for (i in 1:total_number_iterations) {
  print(paste("Progress:", i, "/", total_number_iterations))
  for (j in i:total_number_iterations) 
  {
    correlation_value = cor(cor_of_prism.exp[, i], cor_of_prism.achilles[, j], use = "pairwise.complete.obs")
    correlation_matrix_exp_ach[i, j] = correlation_value
    correlation_matrix_exp_ach[j, i] = correlation_value
  }
}

# Voila!
correlation_matrix_exp_ach
```

```{r}
load("correlation_matrix_exp_ach.RData")
```

Here are some visualization attempts of the correlation matrix between the scores of the gene's expression and their scores in Achilles

```{r}
mini_cor_mat_exp_ach = correlation_matrix_exp_ach[17000:17008,17000:17008]
heatmap(mini_cor_mat_exp_ach, 
        col = colorRampPalette(c("blue", "white", "red"))(100),
        main = "Mini Correlation Matrix",
        xlab = "Genes",
        ylab = "Genes")
```

```{r}
install.packages("pheatmap")
library(pheatmap)
```


```{r}
number_included_sample = 100
sample_genes_for_cormat_indeces = sample(1:17767, number_included_sample)
sample_cormat_exp_ach = correlation_matrix_exp_ach[sample_genes_for_cormat_indeces, sample_genes_for_cormat_indeces]

heatmap_colors = colorRampPalette(c("blue", "white", "red"))(number_included_sample)

# Create the heatmap
pheatmap(sample_cormat_exp_ach, col = heatmap_colors,
         main = "Correlation Matrix Expression-Achilles",
         xlab = "Genes", ylab = "Genes",
         fontsize = 8, border_color = NA,
         colorbar = TRUE, show_rownames = FALSE, show_colnames = FALSE)

```

```{r}
number_included_sample = 500
sample_genes_for_cormat_indeces = sample(1:17767, number_included_sample)
sample_cormat_exp_ach = correlation_matrix_exp_ach[sample_genes_for_cormat_indeces, sample_genes_for_cormat_indeces]

heatmap_colors = colorRampPalette(c("blue", "white", "red"))(number_included_sample)

# Create the heatmap
pheatmap(sample_cormat_exp_ach, col = heatmap_colors,
         main = "Correlation Matrix Expression-Achilles",
         xlab = "Genes", ylab = "Genes",
         fontsize = 8, border_color = NA,
         colorbar = TRUE, show_rownames = FALSE, show_colnames = FALSE)
```

***1. Filter out the relevant correlation values and the relevant anticorrelation values***

```{r}
#Set thresholds for relevance
#Correlation:
threshold_correlation = 0.9 * max(correlation_matrix_exp_ach, na.rm=TRUE)
relevant_correlation_values = correlation_matrix_exp_ach[which(correlation_matrix_exp_ach>threshold_correlation)]
```

```{r}
#Anti-correlation
threshold_anticorrelation = 0.9 * min(correlation_matrix_exp_ach, na.rm=TRUE)
relevant_anticorrelation_values = correlation_matrix_exp_ach[which(correlation_matrix_exp_ach<threshold_anticorrelation)]
```

```{r}
#These values can be found at the following positions in the original correlation_matrix_exp_ach:

#Correlation:
flattened_indices_c = which(correlation_matrix_exp_ach %in% relevant_correlation_values, arr.ind=TRUE)

#Anti-correlation:
flattened_indices = which(correlation_matrix_exp_ach %in% relevant_anticorrelation_values, arr.ind=TRUE)
```

Keep in mind that the prism.achilles data frame captures cell proliferation rates after the Knock Out of the gene of interest. The values describe whether or not after gene KO the proliferation rate of the cell line has increased or decreased. This is inversely proportional to the gene's influence on cell proliferation. E.g. if a gene promotes cell proliferation, a KO of this gene will lead to the cells proliferating at a lower rate, so the prism.achilles score will be LOWER. So LOWER values mean HIGHER relevance in prism.achilles!

```{r}
# In order to access the names of the genes for these values, we have to convert the flattened indices to matrix indices (row and column numbers)

#Correlation:
row_indices_matrix = c()
col_indices_matrix = c()
for (i in 1:length(flattened_indices)){
number_rows_columns = nrow(correlation_matrix_exp_ach)

if(flattened_indices[i] %% number_rows_columns == 0)
{
  row_indices_matrix[i] = flattened_indices[i] %/% number_rows_columns
  col_indices_matrix[i] = flattened_indices[i] %/% number_rows_columns + 1
}else
{
  row_indices_matrix[i] = flattened_indices[i] %% number_rows_columns
  col_indices_matrix[i] = flattened_indices[i] %/% number_rows_columns + 1
}
}
row_indices_matrix
col_indices_matrix

#Anti-correlation:
row_indices_matrix_c = c()
col_indices_matrix_c = c()
for (i in 1:length(flattened_indices_c)){
number_rows_columns = nrow(correlation_matrix_exp_ach)

if(flattened_indices_c[i] %% number_rows_columns == 0)
{
  row_indices_matrix_c[i] = flattened_indices_c[i] %/% number_rows_columns
  col_indices_matrix_c[i] = flattened_indices_c[i] %/% number_rows_columns + 1
}else
{
  row_indices_matrix_c[i] = flattened_indices_c[i] %% number_rows_columns
  col_indices_matrix_c[i] = flattened_indices_c[i] %/% number_rows_columns + 1
}
}
row_indices_matrix_c
col_indices_matrix_c
```
```{r}
#We calculate the so obtained relevant correlation and anti-correlation values

#Correlation
relevant_correlation_values4 = c()
for( i in 1:length(relevant_correlation_values))
{
  relevant_correlation_values4[i] = correlation_matrix_exp_ach[row_indices_matrix_c[i], col_indices_matrix_c[i]]
}
relevant_correlation_values4
```

```{r}
#Anti-correlation
relevant_anticorrelation_values4 = c()
for( i in 1:length(relevant_anticorrelation_values))
{
  relevant_anticorrelation_values4[i] = correlation_matrix_exp_ach[row_indices_matrix[i], col_indices_matrix[i]]
}
relevant_anticorrelation_values4
```

```{r}
#And now we check if we have successfully performed the indices transformation to access the same data:
all(relevant_correlation_values == relevant_correlation_values4)
all(relevant_anticorrelation_values == relevant_anticorrelation_values4)
```
***2. Select the relevant gene pairs that correspond to the afore-mentioned values***

These are the relevant genes involved in the strong correlation/anti-correlation
```{r}
#CORRELATION
relevant_genes_corexpach_correlation = unique(rownames(correlation_matrix_exp_ach[row_indices_matrix_c,]))
relevant_genes_corexpach_correlation
```

```{r}
#ANTICORRELATION
relevant_genes_corexpach_anticorrelation = unique(rownames(correlation_matrix_exp_ach[row_indices_matrix,]))
relevant_genes_corexpach_anticorrelation
```
Let us see how they pair up to form the relevant values we filtered out:

```{r}
#Correlation:
correlation_matrix_exp_ach[row_indices_matrix_c,col_indices_matrix_c]
```

```{r}
#Anti-correlation:
correlation_matrix_exp_ach[row_indices_matrix,col_indices_matrix]
```

```{r}
#Now let's see if our selected values are indeed the diagonals

#Correlation:
copy_cormatexpach_correlation = correlation_matrix_exp_ach[row_indices_matrix_c,col_indices_matrix_c] 
copy_cormatexpach_correlation[which(copy_cormatexpach_correlation == correlation_matrix_exp_ach[flattened_indices_c])] = "RELEVANT"
copy_cormatexpach_correlation
```
```{r}
#Anti-correlation:
copy_cormatexpach_anticorrelation = correlation_matrix_exp_ach[row_indices_matrix,col_indices_matrix] 
copy_cormatexpach_anticorrelation[which(copy_cormatexpach_anticorrelation == correlation_matrix_exp_ach[flattened_indices])] = "RELEVANT"
copy_cormatexpach_anticorrelation
#The reason for the RELEVANT here not to be written only on the diagonals is that some columns are in duplicates.
```

Yes, they are diagonals indeed. This means that:

*i) in correlation pairs: the expression of the one gene is strongly correlated with the proliferation of the cell after the KO of the other gene. E.g. one cell proliferates strongly after CDS1 is knocked out and CDS2 is overexpressed. AND: one cell proliferates weakly after CDS1 is knocked out and CDS2 is underexpressed. This can hint at opposite roles of the two genes and their respective products in cell proliferation.*

Hypothesis model: 
May a cancer cell proliferate strongly when GENE1 is knocked out and have a high expression score of GENE2. This suggests that GENE1 normally inhibits cell proliferation when it isn't knocked out and GENE2 must have a product that promotes cell proliferation, as it is in the cancer cell's interest to upregulate genes that would enhance its spread. So, the effect of GENE1 and GENE2 on cell proliferation must be OPPOSITE.

```{r}
# CORRELATION pairs:
rownames(copy_cormatexpach_correlation)
colnames(copy_cormatexpach_correlation)
```
*INFO:*

This is indeed true for FERMT1 and FERMT2
FERMT1 - negative regulation of stem cell proliferation
https://amigo.geneontology.org/amigo/term/GO:2000647


FERMT2 - positive regulation of stem cell proliferation
https://amigo.geneontology.org/amigo/term/GO:1902462

We actually expected TP53 and EDA2R to have opposite functions. However, their function is similar. Normally, this would contradict our hypothesis, but we found out that these two genes directly influence each other: TP53 activates the transcription of EDA2R. This is consistent with the strong correlation of the data for the two genes:
Brosh R, Sarig R, Natan EB, Molchadsky A, Madar S, Bornstein C, Buganim Y, Shapira T, Goldfinger N, Paus R, Rotter V. p53-dependent transcriptional regulation of EDA2R and its involvement in chemotherapy-induced hair loss. FEBS Lett. 2010 Jun 3;584(11):2473-7. doi: 10.1016/j.febslet.2010.04.058. Epub 2010 Apr 29. PMID: 20434500.
Hypothesis: The expression of one of the genes promotes the expression of the other. So, even though the gene is KO'd, the other one can act as a substitute for the function of the KO'd gene=> upon KO of EDA2R, the cell proliferates weaker because TP53 is still expressed, even if underexpressed. OR: TP53 is KO'd and the cell still proliferates weaker because EDA2R is still expressed, even if underexpressed

RPP25L and RPP25 also interact with each other. However, the effect on cell proliferation is yet unclear:
RPP25 - negative regulation of cell proliferation
Xiao D, Wu J, Zhao H, Jiang X, Nie C. RPP25 as a Prognostic-Related Biomarker That Correlates With Tumor Metabolism in Glioblastoma. Front Oncol. 2022 Jan 12;11:714904. doi: 10.3389/fonc.2021.714904. PMID: 35096558; PMCID: PMC8790702.

RPP25L - function unclear, but interacts with RPP25
Köferle, Anna et al.,Interrogation of cancer gene dependencies reveals paralog interactions of autosome and sex chromosome-encoded genes
Cell Reports, Volume 39, Issue 2, 110636

CDS1 and CDS2 do not have any yet described effect on cell proliferation

With this model, we can predict up to a certain level how two paired genes act in respect to each other: similarly or oppositely. Upon additional information of interaction, we can say whether or not they activate each other.
*=> If the function of specific genes in the group is known, this model can be used to up to a certain extent to predict the functions of the unknown genes they strongly correlate to in the matrix.*


```{r}
plot(new_prism.achilles$TP53)
plot(new_prism.exp$TP53)

plot(new_prism.achilles$EDA2R)
plot(new_prism.exp$EDA2R)

plot(new_prism.achilles$RPP25)
plot(new_prism.exp$RPP25)

plot(new_prism.achilles$RPP25L)
plot(new_prism.exp$RPP25L)
```





*ii) in anti-correlation pairs: the expression of the one gene is strongly anti-correlated with the proliferation of the cell after the KO of the other gene. E.g. one cell proliferates weakly after SOX10 is knocked out and CAPN3 is overexpressed. AND: one cell proliferates strongly after SOX10 is knocked out and CAPN3 is underexpressed. This can hint at similar roles of the two genes and their products in cell proliferation.*

Hypothesis model: 
May a cancer cell proliferate strongly when GENE1 is knocked out and have a low expression score of GENE2. This suggests that GENE1 normally inhibits cell proliferation when it isn't knocked out and GENE2 must have a product that inhibits cell proliferation, as the cancer cell wouldn't promote genes that would inhibit its spread. So, the effect of GENE1 and GENE2 on cell proliferation must be SIMILAR.

```{r}
# ANTICORRELATION pairs:
rownames(copy_cormatexpach_anticorrelation)
colnames(copy_cormatexpach_anticorrelation)
```
*INFO:*

Example of hypothesis for the genes:
SOX10 promotes cancer cell proliferation:
Cronin JC, Watkins-Chow DE, Incao A, Hasskamp JH, Schönewolf N, Aoude LG, Hayward NK, Bastian BC, Dummer R, Loftus SK, Pavan WJ. SOX10 ablation arrests cell cycle, induces senescence, and suppresses melanomagenesis. Cancer Res. 2013 Sep 15;73(18):5709-18. doi: 10.1158/0008-5472.CAN-12-4620. Epub 2013 Aug 1. PMID: 23913827; PMCID: PMC3803156.

Depletion of CAPN3 inhibits cell proliferation (so CAPN3 important for cell proliferation promotion):
"In the absence of CAPN3, immune cell influx is attenuated and expression of pro-myogenic factors is reduced. As a result, cell proliferation and myogenic differentiation is compromised, contributing to a growth deficit under reloading conditions observed in C3KO mice." cited from:
Irina Kramerova and others, Calpain 3 and CaMKIIβ signaling are required to induce HSP70 necessary for adaptive muscle growth after atrophy, Human Molecular Genetics, Volume 27, Issue 9, 01 May 2018, Pages 1642–1653, https://doi.org/10.1093/hmg/ddy071

CPN1 promotes cell proliferation:
Yang S, Yang H, Grisafi P, Sanchatjate S, Fink GR, Sun Q, Hua J. The BON/CPN gene family represses cell death and promotes cell growth in Arabidopsis. Plant J. 2006 Jan;45(2):166-79. doi: 10.1111/j.1365-313X.2005.02585.x. PMID: 16367962.

DCT promotes cell proliferation:
Jiao Z, Zhang ZG, Hornyak TJ, Hozeska A, Zhang RL, Wang Y, Wang L, Roberts C, Strickland FM, Chopp M. Dopachrome tautomerase (Dct) regulates neural progenitor cell proliferation. Dev Biol. 2006 Aug 15;296(2):396-408. doi: 10.1016/j.ydbio.2006.06.006. Epub 2006 Jun 8. PMID: 16857183.

But also: CAPN3 inhibits cell proliferation
Moretti D, Del Bello B, Allavena G, Corti A, Signorini C, Maellaro E. Calpain-3 impairs cell proliferation and stimulates oxidative stress-mediated cell death in melanoma cells. PLoS One. 2015 Feb 6;10(2):e0117258. doi: 10.1371/journal.pone.0117258. PMID: 25658320; PMCID: PMC4319969.
=> result may still be unclear at some points

```{r}
plot(new_prism.achilles$SOX10)
plot(new_prism.exp$SOX10)

plot(new_prism.achilles$CAPN3)
plot(new_prism.exp$CAPN3)

plot(new_prism.achilles$CPN1)
plot(new_prism.exp$CPN1)

plot(new_prism.achilles$DCT)
plot(new_prism.exp$DCT)
```

```{r}
new_prism.exp$SOX10[which(new_prism.exp$SOX10 >6)[which(new_prism.achilles$CAPN3[which(new_prism.exp$SOX10 >6)] < 0.9*min(new_prism.achilles$CAPN3[which(new_prism.exp$SOX10 >6)],na.rm=TRUE))]]
new_prism.achilles$CAPN3[which(new_prism.exp$SOX10 >6)[which(new_prism.achilles$CAPN3[which(new_prism.exp$SOX10 >6)] < 0.9*min(new_prism.achilles$CAPN3[which(new_prism.exp$SOX10 >6)],na.rm=TRUE))]]

#This line of code shows that even for the outliers of the gene expression of SOX10, the outliers of the proliferation after KO of CAPN3 match in magnitude
```
*=> If the function of specific genes in the group is known, this model can be used to up to a certain extent to predict the functions of the unknown genes they strongly anti-correlate to in the matrix.*


***3. Inspect the correlation and anticorrelation of the genes with each other***

In order to see which gene's expression strongly correlates or anti-correlates with the cell's proliferation behaviour after their knockout, we need to consider the diagonal values of the computed correlation matrix
```{r}
diag_genes_cor_exp_ach = diag(correlation_matrix_exp_ach)
diag_genes_cor_exp_ach
```

Selection of relevant values...
```{r}
threshold_correlation90 = 0.9 * max(correlation_matrix_exp_ach, na.rm=TRUE)
threshold_anticorrelation90 = 0.9 * min(correlation_matrix_exp_ach, na.rm=TRUE)

relevant_correlation_values = which(diag_genes_cor_exp_ach > threshold_correlation90)
relevant_anticorrelation_values = which(diag_genes_cor_exp_ach< threshold_anticorrelation90)

relevant_correlation_values
relevant_anticorrelation_values
```
Lowering the threshold...
```{r}
threshold_correlation80 = 0.8 * max(correlation_matrix_exp_ach, na.rm=TRUE)
threshold_anticorrelation80 = 0.8 * min(correlation_matrix_exp_ach, na.rm=TRUE)

relevant_correlation_values = which(diag_genes_cor_exp_ach > threshold_correlation80)
relevant_anticorrelation_values = which(diag_genes_cor_exp_ach< threshold_anticorrelation80)

relevant_correlation_values
relevant_anticorrelation_values
```
```{r}
threshold_correlation75 = 0.75 * max(correlation_matrix_exp_ach, na.rm=TRUE)
threshold_anticorrelation75 = 0.75 * min(correlation_matrix_exp_ach, na.rm=TRUE)

relevant_correlation_values = which(diag_genes_cor_exp_ach > threshold_correlation75)
relevant_anticorrelation_values = which(diag_genes_cor_exp_ach< threshold_anticorrelation75)

relevant_correlation_values
relevant_anticorrelation_values
```
ANTICORRELATION:

The anti-correlation between the expression and the Achilles scores could be interpreted as following:

i) GENE is under-expressed in the cancer cell line and the cancer cell line proliferates stronger when GENE is knocked out. This hints at the character of GENE to be that of a TUMOR SUPPRESSOR GENE. A tumor suppressor gene would be under-expressed in cancer cells and cancer cells would proliferate stronger if there is no tumor suppressor gene blocking their prosperity.

OR:

ii) GENE is overexpressed in the cancer cell line and the cancer cell line proliferates at a lower rate when GENE is knocked out. This hints at the character of GENE to be that of an ONCOGENE. Oncogenes would be over-expressed in cancer cells and cancer cells would not proliferate as well if there would be no active oncogene promoting their spread.

By knowing the values of one of the two, either the gene's expression or it's Achilles score, we could predict the character of that gene:
```{r}
plot(new_prism.achilles$HNF1B)
mean(new_prism.achilles$HNF1B, na.rm=TRUE)

plot(new_prism.achilles$IRF4)
mean(new_prism.achilles$IRF4, na.rm=TRUE)

plot(new_prism.achilles$PAX8)
mean(new_prism.achilles$PAX8, na.rm=TRUE)

plot(new_prism.achilles$SOX10)
mean(new_prism.achilles$SOX10, na.rm=TRUE)

plot(new_prism.achilles$TP63)
mean(new_prism.achilles$TP63, na.rm=TRUE)

```

Prediction: HNF1B = Oncogene
Truth:      HNF1B = Tumor Suppressor Gene
Lu W, Sun J, Zhou H, Wang F, Zhao C, Li K, Fan C, Ding G, Wang J. HNF1B inhibits cell proliferation via repression of SMAD6 expression in prostate cancer. J Cell Mol Med. 2020 Dec;24(24):14539-14548. doi: 10.1111/jcmm.16081. Epub 2020 Nov 10. PMID: 33174391; PMCID: PMC7754016.

Prediction: IRF4 = Oncogene
Truth:      IRF4 = Both Tumor Suppressor Gene and Oncogene
Maffei, R., Fiorcari, S., Atene, C.G. et al. The dynamic functions of IRF4 in B cell malignancies. Clin Exp Med (2022). https://doi.org/10.1007/s10238-022-00968-0

Prediction: PAX8 = Oncogene
Truth:      PAX8 = Oncogene
Bie LY, Li N, Deng WY, Lu XY, Guo P, Luo SX. Evaluation of PAX8 expression promotes the proliferation of stomach Cancer cells. BMC Mol Cell Biol. 2019 Dec 27;20(1):61. doi: 10.1186/s12860-019-0245-9. Retraction in: BMC Mol Cell Biol. 2021 Jun 01;22(1):31. PMID: 31881968; PMCID: PMC6935224.

Prediction: SOX10 = Oncogene
Truth:      SOX10 = Oncogene (possible)
Cronin JC, Watkins-Chow DE, Incao A, Hasskamp JH, Schönewolf N, Aoude LG, Hayward NK, Bastian BC, Dummer R, Loftus SK, Pavan WJ. SOX10 ablation arrests cell cycle, induces senescence, and suppresses melanomagenesis. Cancer Res. 2013 Sep 15;73(18):5709-18. doi: 10.1158/0008-5472.CAN-12-4620. Epub 2013 Aug 1. PMID: 23913827; PMCID: PMC3803156.

Prediction: TP63 = Tumor Suppressor Gene
Truth:      TP63 = Tumor SUppressor Gene 
Melino, G. p63 is a suppressor of tumorigenesis and metastasis interacting with mutant p53. Cell Death Differ 18, 1487–1499 (2011). https://doi.org/10.1038/cdd.2011.81

*=> This model could predict the main function of a gene regarding cell proliferation and tumorigenesis (up to a certain extent)*

CORRELATION:

We see no relevant correlation bacause a correlation for the these datasets would mean contradicting effects on cell proliferation for the same genes.
(1 gene at 70% LTO1?)


```{r}
pheatmap(copy_cormatexpach_correlation)
copy_cormatexpach_anticorrelation = correlation_matrix_exp_ach[row_indices_matrix,col_indices_matrix]
pheatmap(copy_cormatexpach_anticorrelation)
```

